### PR TITLE
[jit] Enable testing on Loss modules

### PR DESCRIPTION
**Pull request by driazati**
This PR adds `None` buffers as parameters (similarly to #14715). It also cleans up a bunch of the `test_jit.py` tests that should be covered by `common_nn.py` and brings in `criterion_tests` to test loss functions.

### CODE REVIEW

File: torch/jit/__init__.py [hunk 1]
```diff
@@ -1159,7 +1159,10 @@ def __init__(self, original, stubs):
                 elif isinstance(item, Parameter) or (isinstance(item, Module) and item is not self):
                     ScriptModule.__setattr__(self, name, item)
             for name in original._buffers:
-                self.register_buffer(name, original._buffers[name])
+                if original._buffers[name] is None:
```
Comment (wanchaol) [inline torch/jit/__init__.py]: can you make this the same semantic with parameters, or make parameters same semantic as this

File: test/test_jit.py [hunk 2]
```diff
@@ -10390,65 +10378,6 @@ def func(x):
...
-
-
-def to_module_test_format(tup):
-    dic = dict(module_name=tup[0], constructor_args=tup[1], input_fn=lambda: tup[2])
-    if len(tup) >= 5:
-        dic['desc'] = tup[4]
-    local_module_tests.append(dic)
-
-
-def add_interpolate_module_tests():
```
Comment (eellison) [inline test/test_jit.py]: Are these still being tested ?

Comment (driazati) [reply test/test_jit.py]: Yeah there are 20+ `Upsample` tests [here](https://github.com/pytorch/pytorch/blob/master/test/common_nn.py#L1673)

**Review by eellison — APPROVED** [review]
Wait for tests to pass

File: torch/jit/__init__.py (hunk 1 above)
Comment (driazati) [reply torch/jit/__init__.py]: Will do in a follow up PR just to get things landed faster

File: test/test_jit.py [hunk 3]
```diff
@@ -9843,27 +9843,11 @@ def forward(self, x, y):
 }
 
 EXCLUDE_SCRIPT_MODULES = {
+    'test_nn_NLLLoss_weights_ignore_index_neg',
```
Comment (eellison) [inline test/test_jit.py]: Why was this added ?

Comment (driazati) [reply test/test_jit.py]: It looks like it's failing in aten for some case that wasn't being tested before
